### PR TITLE
feat: release of infinitemonkeys helm chart

### PR DESCRIPTION
**Pull request by jschumacher-wire**
feat: release of infinitemonkeys helm chart 

WPB-2595 / WPB-6448

* adding readme
* minor refactor
* removing nested docker install from Dockerfile

This PR adds a helm chart for Infinite Monkeys, enabling native K8s deployments with any number of test users (eg. monkeyServer pods).

### CODE REVIEW

**Review by augustocdias — APPROVED** [review]

File: monkeys/helm/infinitemonkeys/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: v2
+
+name: infinitemonkeys
+type: application
+version: 0.1.0
+
+dependencies:
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/Chart.yaml:7]: please remove the dependency

File: monkeys/helm/infinitemonkeys/Readme.md [hunk 2]
```diff
@@ -0,0 +1,30 @@
+# infinitemonkeys - load testing tool for Wire
+
+This Helm chart provides:
+
+- A group of StatefulSets deploying one monkeyController and N monkeyServers
+- Integration with Prometheus and Grafana via dependency "kube-prometheus-stack"
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/Readme.md:6]: - Integration with Prometheus-Operator via PodMonitors and a Grafana Dashboard

File: monkeys/helm/infinitemonkeys/templates/configmaps.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+apiVersion: v1
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/templates/configmaps.yaml:1]: Please introduce a values parameter, that enables or disables the monitoring integration - subsequently then creates the dashboard and podmonitors if set to "true"

File: monkeys/helm/infinitemonkeys/templates/podmonitor.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: "{{ .Values.service.serverName }}-podmonitor"
+  namespace: {{ .Values.namespace }}
+spec:
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/templates/podmonitor.yaml:6]: please add the standard helm labels
  labels:
    chart: "{{ .Chart.Name }}-{{ .Chart.Version }}"
    release: "{{ .Release.Name }}"

File: monkeys/helm/infinitemonkeys/templates/podmonitor.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+  namespaceSelector:
+    any: true
+
+---
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
+metadata:
+  name: "{{ .Values.service.controllerName }}-podmonitor"
+  namespace: {{ .Values.namespace }}
+spec:
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/templates/podmonitor.yaml:22]: also here

File: monkeys/helm/infinitemonkeys/values.yaml [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  name: Diya
+  teamName: Diya
+  instanceURL: diya.wire.link
+  userCount: 3
+  authUser: basic-auth-user
+  authPassword: "..."
+
+# replicaCount needs to be one more than userCount above
+# as monkeyController expects an additional owner
+monkeyServer:
```
Comment (mastaab) [inline monkeys/helm/infinitemonkeys/values.yaml:16]: maybe this can be calculated with userCount +1 ?
so we don't need an additional parameter

**Review by mastaab — CHANGES_REQUESTED** [review]

**Review by mastaab — APPROVED** [review]

File: monkeys/helm/infinitemonkeys/templates/monkey-controller.yaml [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+          command: ['bash', '-c']
+          args:
+            {{- include "service.waitMonkeyServerReady" . | indent 10 }}
+      containers:
+        - name: monkeycontroller
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
+          command:
+            - /opt/app/bin/monkeys
+          args:
+            - '-d1500'
```
[the remark this answers is not in the document]
Comment (mastaab) [reply monkeys/helm/infinitemonkeys/templates/monkey-controller.yaml:49]: they serve different purposes
the -d is to delay the time for a new conversation to be added to the pool for other monkeys to use
this is to reduce the conversation not found errors

**Review by mastaab — APPROVED** [review]

**Pull request merged**